cors.go

```diff
@@ -71,7 +166,17 @@ func parseCORSOptions(opts ...CORSOption) *cors {
 // The headers Content-Type, Expires, Cache-Control, ... are always allowed.
 func AllowedHeaders(headers []string) CORSOption {
 	return func(ch *cors) error {
-		ch.allowedHeaders = headers
+		for _, v := range headers {
+			normalizedHeader := http.CanonicalHeaderKey(strings.TrimSpace(v))
+			if normalizedHeader == "" {
+				continue
+			}
+
+			if !ch.isMatch(normalizedHeader, ch.allowedHeaders) {
```

Comment:
Good move—definitely need to retain the default headers when adding custom allowed headers.